.github/workflows/main.yml

```
@@ -92,3 +92,25 @@ jobs:
       # The Windows CI transforms the `c-library` symlink into a real directory, modifying the Git
       # state, so we ignore these changes with `--allow-dirty` here.
       run: scripts/verify-publish.sh --allow-dirty
+
+  itt_python_build_and_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+          - os: windows-latest
+    runs-on: ['${{matrix.os}}']
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v1
```

```suggestion
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```
actions/checkout@v1 is deprecated